packages/governance/src/contractGovernance/governApi.js

```diff
@@ -0,0 +1,132 @@
+} from '../question.js';
+
+const { details: X } = assert;
+
+/**
+ * Make a pair of positions for a question about whether to invoke an API. If
+ * the vote passes, the method will be called on the governedApis facet with the
+ * parameters that were provided.
+ *
+ * @param {string} apiMethodName
+ * @param {[unknown]} methodParams
+ */
+const makeApiInvocationPositions = (apiMethodName, methodParams) => {
+  const positive = harden({ apiMethodName, methodParams });
+  const negative = harden({ dontInvoke: apiMethodName });
+  return { positive, negative };
+};
+
+/**
+ * manage contracts that allow governance to invoke functions.
+ *
+ * @param {ERef<ZoeService>} zoe
+ * @param {Instance} governedInstance
+ * @param {any} governedApis
+ * @param {string[]} governedNames
+ * @param {ERef<TimerService>} timer
+ * @param {() => Promise<PoserFacet>} getUpdatedPoserFacet
+ * @returns {Promise<ApiGovernor>}
+ */
+const setupApiGovernance = async (
+  zoe,
+  governedInstance,
+  governedApis,
+  governedNames,
+  timer,
+  getUpdatedPoserFacet,
+) => {
+  /** @type {WeakSet<Instance>} */
+  const voteCounters = new WeakSet();
+
+  /** @type {VoteOnApiInvocation} */
+  const voteOnApiInvocation = async (
+    apiMethodName,
+    methodParams,
+    voteCounterInstallation,
+    deadline,
+  ) => {
+    const outcomeOfUpdateP = makePromiseKit();
+
+    assert(
+      governedNames.includes(apiMethodName),
+      X`${apiMethodName} is not a governed API.`,
+    );
+
+    const { positive, negative } = makeApiInvocationPositions(
+      apiMethodName,
+      methodParams,
+    );
+
+    /** @type {ApiInvocationIssue} */
+    const issue = harden({ apiMethodName, methodParams });
+    const questionSpec = coerceQuestionSpec({
+      method: ChoiceMethod.UNRANKED,
+      issue,
+      positions: [positive, negative],
+      electionType: ElectionType.API_INVOCATION,
+      maxChoices: 1,
+      closingRule: { timer, deadline },
+      quorumRule: QuorumRule.MAJORITY,
+      tieOutcome: negative,
+    });
+
+    const { publicFacet: counterPublicFacet, instance: voteCounter } = await E(
+      getUpdatedPoserFacet(),
+    ).addQuestion(voteCounterInstallation, questionSpec);
+
+    voteCounters.add(voteCounter);
+
+    // CRUCIAL: Here we wait for the voteCounter to declare an outcome, and then
+    // attempt to invoke the API if that's what the vote called for. We need to
+    // make sure that outcomeOfUpdateP is updated whatever happens.
+    //
+    // * If the vote was negative, resolve to the outcome
+    // * If the API invocation succeeds, say so
+    // * If the update fails, reject the promise
+    // * if the vote outcome failed, reject the promise.
+    E(counterPublicFacet)
+      .getOutcome()
+      .then(outcome => {
+        if (keyEQ(positive, outcome)) {
+          E(governedApis)
+            [apiMethodName](...methodParams)
+            .then(returnValue => outcomeOfUpdateP.resolve(returnValue))
+            .catch(e => {
```

Comment:
Four cases defined here. If we use regular promises instead of promisekit, a throw will automatically reject. So think think this can be simplified to:
```
    const outcomeOfUpdate = E(counterPublicFacet)
      .getOutcome()
      .then(outcome => {
        if (keyEQ(positive, outcome)) {
          return E(governedApis)[apiMethodName](...methodParams);
        } else {
          return negative;
        }
      });
```
Now it's much easier for me to see that it resolves to either the return value of the method call or an object representing the position of a negative outcome. This is confusing. The term "outcome" is used here to mean "positive or negative" but "outcomeOfUpdate" returns "negative or the return value from executing the positive outcome".

What are the use cases for getting a result back from the method call? Since it's something people vote on executing, it seems like you would only execute "void" methods. In which case all the caller needs to know is a) whether the vote went positive or negative and b) that when they get this answer that if it was positive then that call is completed. IIUC, alll the other states are exceptional (e.g. failure to make remote call)